### ISSUE TITLE
Can I use a custom metrics on model.fit? 

### ISSUE THREAD

**Issue by leocd91** [lRomul/argus#6]
Can I use custom metrics? 
Also if not, do you have any suggestions on metrics for Multiclass Segmentation problems?

**Comment by lRomul** [comment]
Yes, you can find a section on metrics in the `dev` version of the documentation 
https://pytorch-argus.readthedocs.io/en/dev/metrics.html

I am working on more detailed docs. Thank you for using the library!

**Comment by lRomul** [comment]
Here is an example of metric implementation from a segmentation Kaggle competition. 
Metric description: https://www.kaggle.com/c/tgs-salt-identification-challenge/overview/evaluation
Implementation: https://github.com/lRomul/argus-tgs-salt/blob/master/src/metrics.py

**Comment by leocd91** [comment]
Thanks. It's a nice lib btw.

**Comment by leocd91** [comment]
Hi, I have an Issue where the preds and targets got different shape.. 

here's my code on modified metrics :
```
class multiclassf1(Metric):
    """Calculates the accuracy for multiclass classification."""

    name = 'multif1'
    better = 'max'

    def __init__(self):
        self.score = 0

    def reset(self):
        self.score = 0

    def update(self, step_output: dict):
        preds = step_output['prediction'].cpu().numpy()
        print(preds.shape)
        trgs = step_output['target'].cpu().numpy()
        print(trgs.shape)
        self.score =  compute_scores(trgs, preds)

    def compute(self):

        return self.score
```
the **preds** shape is (2, 6, 512, 256), where the **trgs** is (2, 512, 256). (example batchsize=2, that's why it's 2 in the front)
It's a semantic segmentation with 6 classes.
any suggestion?

**Comment by leocd91** [comment]
ah, solved it by changed it to `preds = step_output['prediction'].max(1)[1].cpu().numpy()[:, :, :]`

**Comment by lRomul** [comment]
Hi, glad you solved the problem!
It seems you are using CrossEntropyLoss or NLLLoss. They are work on targets with int values that represent class indexes.

**Comment by leocd91** [comment]
I love the monitor feature so far! 
Can I custom it to make a pic of `pred` vs `target` everytime there's an improvement score in a epoch?

**Comment by lRomul** [comment]
@leocd91, hi!
Hi! You can create a custom `MonitorCheckpoint` that will save `step_output` of the last iteration in the epoch with an improved score. 

``` python
import torch
from argus.engine import State
from argus.utils import deep_to
from argus.callbacks import MonitorCheckpoint


class CustomMonitorCheckpoint(MonitorCheckpoint):
    def iteration_complete(self, state: State):
        # save a step_output after each iteration
        self.step_output = deep_to(state.step_output,
                                   device='cpu')

    # override checkpoint saving method
    def save_model(self, state: State, file_path):
        state.model.save(file_path)  # save model
        torch.save(self.step_output,   # save step_output
                   f'step_output_{state.epoch}.pth')
```
